Assets/BossRoom/Scripts/Client/Net/ClientSessionManager.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI;
+using UnityEngine;
+using MLAPI.SceneManagement;
+
+namespace BossRoom.Client
+{
+    public class ClientSessionManager : MonoBehaviour
+    {
+        public GameNetPortal m_Portal;
+
+        private ClientGameNetPortal m_ClientGameNetPortal;
+
+        private HashSet<ulong> m_ClientIDs = new HashSet<ulong>();
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            if (m_Portal == null)
+            {
+                gameObject.SetActive(false);
+                return;
+            }
+
+            m_ClientGameNetPortal = m_Portal.GetComponent<ClientGameNetPortal>();
+
+            m_Portal.NetManager.OnClientConnectedCallback += OnClientConnected;
+            m_Portal.NetManager.OnClientDisconnectCallback += OnClientDisconnected;
+
+            DontDestroyOnLoad(this);
+        }
+
+        private void OnDestroy()
+        {
+            m_Portal.NetManager.OnClientConnectedCallback -= OnClientConnected;
+            m_Portal.NetManager.OnClientDisconnectCallback -= OnClientDisconnected;
+        }
+
+        private void OnClientConnected(ulong clientID)
+        {
+            if (!m_ClientIDs.Contains(clientID))
```

Comment:
Issue with ClientIDs is that they are reused if players reconnect. So clientA and clientB could have client IDs 1 and 2. if both disconnect and reconnect, clientA could have id 2 and clientB could have id 1.
We already have a session guid used to save your character class, I would have consolidated those into your current solution (which seems cleaner :) )

Reply:
I'm not sure how to get the clientGUIDs, I tried changing it to OnConnectionApproval and processing the payload, but I'm not sure how to get that info.